api/v1alpha1/workspace_types.go

```diff
@@ -23,10 +23,32 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type StorageSpec struct {
+	// Persistent volume claim name
+	PVCName string `json:"pvcName,omitempty"`
+	// Kubernetes storage class to use
+	StorageClass string `json:"storageClass,omitempty"`
+	// Size of the storage
+	Size string `json:"size,omitempty"`
+	// Define the EFS storage
+	AWSEFS AWSEFSSpec `json:"awsEFS,omitempty"`
+}
+
+type StorageStatus struct {
+	AWSEFS AWSEFSStatus `json:"awsEFS,omitempty"`
+}
+
 // WorkspaceSpec defines the desired state of Workspace
 type WorkspaceSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// The username of the user
+	Username string `json:"username,omitempty"`
```

Comment:
Also here, with the Username and Namespace, do we need to let this be specified or could we generate (or default to) a predictable name calculated from the workspace name?

Reply:
yup, sure. I'm going to make a counterpart python generator script to generate the manifests from a config file next. For now it will just rely on users running and committing the changes, but eventually we can hook it into pulsar messages for new users. I was going to let that script fill in sensible defaults for parameters that aren't specifically set.